BrightID/src/components/Connections/ConnectionScreen.tsx

```diff
@@ -58,14 +64,17 @@ function ConnectionScreen(props: Props) {
     connection,
     verificationsTexts,
     connectedAt,
-    mutualGroups,
     mutualConnections,
+    mutualGroups,
     loading,
   } = props;
   const navigation = useNavigation();
+  const myConnections = useSelector(selectAllConnections);
 
   const [groupsCollapsed, setGroupsCollapsed] = useState(true);
   const [connectionsCollapsed, setConnectionsCollapsed] = useState(true);
+  const [possibleDuplicatesCollapsed, setPossibleDuplicatesCollapsed] = useState(true);
+  const possibleDuplicates = myConnections.filter(conn => stringSimilarity(conn.name, connection.name) >= 0.6 && conn.id != connection.id);
```

Comment:
This will evaluate similar connections on each render of the component. I fear that this will cause performance problems for brightID stars with huge number of connections (they have several thousand connections). 
We should move searching for duplicates inside a useEffect() block, so it is not running on each render.

Reply:
Fixed